tests/DPDK_Flow_Dump_with_action_selector.py

```diff
@@ -0,0 +1,156 @@
+        config["relax"] = True
+
+        test_params = test_params_get()
+        config_json = test_params["config_json"]
+        self.config_data = get_config_dict(config_json)
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
+        self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
+
+    def runTest(self):
+        # Compile p4 file using p4c compiler and generate binary using ovs pipeline builder
+        if not test_utils.gen_dep_files_p4c_tdi_pipeline_builder(self.config_data):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+
+        # Create ports using gnmi-ctl
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to configure gnmi ctl ports")
+
+        # Set ip address for interface
+        ip_set_ipv4(self.interface_ip_list)
+
+        # Run Set-pipe command for set pipeline
+        if not p4rt_ctl.p4rt_ctl_set_pipe(
+            self.config_data["switch"],
+            self.config_data["pb_bin"],
+            self.config_data["p4_info"],
+        ):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+
+        table = self.config_data["table"][0]
+
+        # Add action profile members
+        log.info("Add action profile members")
+        for member in table["member_details"]:
+            if not p4rt_ctl.p4rt_ctl_add_member_and_verify(
+                table["switch"], table["name"], member
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add member {member}")
+
+        # Adding action selector groups
+        log.info("Adding action selector groups")
+        group_count = 0
+        for group in table["group_details"]:
+            if not p4rt_ctl.p4rt_ctl_add_group_and_verify(
+                table["switch"], table["name"], group
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add group {group}")
+            group_count += 1
+
+        # Setting rule for l3 action selector
+        log.info(f"Setting up rule for : {table['description']}")
+        table = self.config_data["table"][1]
+        for match_action in table["match_action"]:
+            if not p4rt_ctl.p4rt_ctl_add_entry(
+                table["switch"], table["name"], match_action
+            ):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+      
+        # Get dump entries table
+        dump_table = p4rt_ctl.p4rt_ctl_dump_entities(self.config_data['switch'])
+        # Verify each entry
+        if len(dump_table[1:-1]) != len(self.config_data["flow_dump_table"]):
+            log.failed(f"The table {dump_table[1:-1]} has different entry than definition")
```

Comment:
same as above

Reply:
The statements have a variables {entry}. The f is needed